internal/pkg/manifest/pipeline.go

```diff
@@ -86,9 +86,11 @@ type Source struct {
 }
 
 // PipelineStage represents configuration for each deployment stage
-// of an application.
+// of a workspace. A stage consists of the Archer Environment the pipeline
+// is deloying to and the containerized applications that will be deployed.
 type PipelineStage struct {
-	Name string `yaml:"env"`
+	*archer.Environment `yaml:",inline"`
```

Comment:
Is there a typo in the tag value `yaml:",inline"`, specifically the `,`?

Reply:
> Hm, I don't mind a little duplication if these are truly different, which they seem to be. A Pipeline stage happens to be associated with an environment, but they aren't an environment. The yaml/json mixing is confusing and will stop us from ever serializing the envs to yaml in SSM.
> 
> It's quite common to have different views of the same logical resource represented differently within a go project. Take a look at [this](https://github.com/katzien/go-structure-examples/tree/master/domain-hex/pkg).

I renamed the `Environment` struct in the pipeline.go to `Stages` because like you say, an Archer environment and a Pipeline stage are quite different. From the current progress I made on pipeline CFN generation, I found at least the `Project`, `Name`, `AccountId`, `Prod` to be reusable but not`Region` nor `RegistryURL`. Will keep the environment and "association with an environment" separate for now.

Reply:
@sonofachamp flags have to be separated by `,`. Otherwise goyaml treats it as if the name of the field is "inline" lol